### PR TITLE
DM-38777:  LinearizeSpline linearity corrections do not anchor the spline at zero flux

### PR DESCRIPTION
**Pull request by czwa**
Anchor spline linearity corrections.

### CODE REVIEW

File: python/lsst/cp/pipe/linearity.py [hunk 1]
```diff
@@ -453,6 +453,15 @@ def run(self, inputPtc, dummy, camera, inputDims, inputPhotodiodeData=None,
                     binCenters = binCenters[numPerBin > 0]
 
                 self.debugFit('splineFit', binCenters, np.abs(values), values, None, ampName)
+                # Anchor the spline to have zero correction at zero
+                # flux as well as at the lowest measured flux bin.
+                if binCenters[0] != 0.0:
```
Comment (taranu) [inline python/lsst/cp/pipe/linearity.py:458]: Should this check if any binCenters are negative?

File: python/lsst/cp/pipe/linearity.py [hunk 2]
```diff
@@ -453,6 +453,15 @@ def run(self, inputPtc, dummy, camera, inputDims, inputPhotodiodeData=None,
...
+                    if values[0] != 0.0:
```
Comment (taranu) [inline python/lsst/cp/pipe/linearity.py:459]: Does this behave appropriately in both cases of `values[0]<0` and `values[0]>0`? Just wondering if you can end up with negative `values` and if that's worth checking or if that's only possible if you did something obviously wrong in the first place.

Comment (czwa) [reply python/lsst/cp/pipe/linearity.py:462]: Negative values in the correction is fine; the `values` correspond to amounts to be added to pixels at that flux, and negative values simply means some amount needs to be subtracted.

File: python/lsst/cp/pipe/linearity.py (hunk 1 above)
Comment (czwa) [reply python/lsst/cp/pipe/linearity.py:461]: There shouldn't be any negative binCenters, as these are measured from a series of flat field images.  I've added a `raise` if there are negative binCenters, as that means something has gone wrong.

**Review by taranu — APPROVED** [review]

**Pull request merged**